apps/docs/docs/cdk/render-strategies/rx-schedule-task.md

```diff
@@ -0,0 +1,112 @@
+# rxScheduleTask
+
+`rxScheduleTask` provides a helper function to schedule function execution. It is a minimal building block for making performance optimizations in your code.
+
+## Motivation
+
+Chromium based browsers consider all tasks that take more than 50ms as long tasks. If task runs for more than 50ms, users will start noticing lags. Optimally all user interactions should happen at 30 fps frame-rate with 32ms budget per browser task. In an ideal world, it should be 60 fps and 16ms budget.
+
+> 💡 In reality browser has a reserved overhead of 4ms, try to stick to 28ms of work for 30 fps and 12ms for 60 fps.
+
+## Scheduling mechanisms in browser
+
+Most common ways of delaying a task execution are:
+
+- `setTimeout`
+- `requestAnimationFrame`
+- `requestIdleCallback`
+
+`rxScheduleTask` provides a similar API but comes with huge benefits of notion of frame budget and priority configuration.
+
+## Concurrent strategies
+
+> 💡 Under the hood all our concurrent strategies are based on MessageChannel technology.
+
+To address the problem of long tasks and help browser split the work @rx-angular/cdk provides concurrent strategies. These strategies will help browser to chunk the work into non-blocking tasks whenever it's possible.
+
+You can read detailed information about concurrent strategies [here](strategies/concurrent-strategies.md).
+
+## Usage examples
+
+### Input params
+
+- Just as common delaying apis this method `accepts` a work function that should be scheduled.
+- It also accepts configuration object as an optional second parameter
+  - `strategy` which will be used for scheduling (`normal` is default, for full list of available strategies see [concurrent strategies documentation](strategies/concurrent-strategies.md))
+  - `delay` which is responsible for delaying the task execution (default is 0ms)
+  - `ngZone` if you want your function be executed withing ngzone (default scheduling runs out of zone)
+
+### Return type
+
+Function returns a callback that you can use to cancel already scheduled tasks.
+
+### Default usage
+
+```typescript
+import { rxScheduleTask } from '@rx-angular/cdk/render-strategies';
```

Comment:
```suggestion
import { rxScheduleTask } from '@rx-angular/cdk/render-strategies';
```

I don't like it that we export it from render-strategies. Because it doesn't do anything with rendering.

Reply:
Can you think about better place? We had this discussion in old PR and decided that this is the best option we currently have. We can say the same about RxStrategyProvider, aren't we? This is a service for scheduling any type of work, but it's still in the render-strategies. Same reasoning goes to rxScheduleTask. 

You can make it "related" to rendering by doing rxScheduleTask(() => this.cdRef.detectChanges()) ¯\_(ツ)_/¯.

Reply:
we can introduce a new sub-package for `scheduling`?

Reply:
I'd go for it!

Reply:
hmmm

Reply:
I would do a rethink before and maybe see what other code we want to move

Reply:
Whatever we decide, please let's not have it frozen for another 1.5 years :D

Reply:
+1 for `scheduling`